tensorflow/compiler/mlir/lite/ir/tfl_ops.td

```diff
@@ -1962,6 +1962,25 @@ def TFL_LogOp: TFL_Op<"log", [
   let hasFolder = 1;
 }
 
+def TFL_TableOp: TFL_Op<"table", [
```

Comment:
Is this file needed? It's not very clear what's the end-to-end workflow with a dedicated Table operator.

Reply:
Yes, we need to be able to create a `mlir::TFL::TableOp` in the the converter. You can check the [proof-of-concept](https://github.com/Tessil/tensorflow/commit/84eedece6d8d409a040362a375d669a3a0997abb#diff-9f4911916593feb3005c005582353a39430b45e161458a36673ea1bfac78b26fR106) I linked above that demonstrates the expected end-to-end workflow. We convert a `mlir::TFL::ExpOp` with a quantized input/output range to a `mlir::TFL::TableOp`.

Reply:
Thanks for the end-to-end example. A high level question: LUT is replacing reference and optimized implementation?

Reply:
Yes as there is no other way currently. The choice between reference or optimized kernels is decided at runtime when creating the `tf.lite.Interpreter` (a same *.tflite file can be run with the reference or optimized kernels without problem) . On the other hand, the `TABLE` operator is created during the generation of the *.tflite file we thus can't have a different implementation for the initial converted operator once it's done.